src/modules/anime.js

```diff
@@ -1,4 +1,38 @@
-const _anitomyscript = require('anitomyscript') 
+const _anitomyscript = require('anitomyscript'); 
+const fastLevenshtein = require('fast-levenshtein');
+const { API_BASE_URL } = require('../constants/config');
+const { normalize } = require('./utils');
+
+const processAnimes = async (rssData) => {
+  const animeTitles = rssData.map(anime => anime.title);
+  const parsedAnimes = await anitomyscript(animeTitles);
+
+  const response = await fetch(`${API_BASE_URL}/anime/search`, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ animes: parsedAnimes.map(a => a.anime_title) })
+  });
+  const animes = await response.json()
+
+  const resolvedData = await Promise.all(animes.map(async (anime) => {
+    const bestMatch = parsedAnimes.reduce((best, parsed) => {
+      const distance = fastLevenshtein.get(
+        normalize(anime.title.romaji),
+        normalize(parsed.anime_title)
+      );
+      return distance < best.distance ? { parsed, distance } : best;
+    }, { distance: Infinity }).parsed;
+
+    if (bestMatch && bestMatch.episode_number) {
+      const episodeData = await fetch(`${API_BASE_URL}/anime/${anime.idAnil}/${bestMatch.episode_number}`).then(res => res.json());
+      const rssTorrent = rssData.find(anim => anim.title === bestMatch.file_name);
+      return { ...anime, episode: episodeData, torrent: rssTorrent };
+    }
+    return null;
+  }));
+
+  return resolvedData.filter(Boolean);
+}
```

Comment:
Haz mas legible el codigo, tipo por ejemplo son muchos pasos logicos, te recomiendo separar cada paso logino en un nombre de variable que describa ese paso, tipo si haces un codigo que su proposito es buscar el mejor match, asi como usas `bestMatch` esta perfecto pero es dificil entender la logica, tienes muchas cosas en la siguiente linea:

```js
  const resolvedData = await Promise.all(animes.map(async (anime) => {
  ...
```
Igual si se entiende pero es mas recomendable en un array guardar ahi las promesas, para fragmentar un poco el codigo y sea mas facil de leer, tipo:

```js
  const animeDataPromises = animes.map(async (anime) => {
  ....
```

Y luego le haces ya un await all:

```js
  const animeData = await Promises.all(animeDataPromises);
  ```
  
  Ahi ya separamos un poco la parte de crear las promesas y la de esperarlas a todas.
  
  ---
  Ahora, esta callback function esta muy dificil de leer:
  ```js
  async (anime) => {
    const bestMatch = parsedAnimes.reduce((best, parsed) => {
      const distance = fastLevenshtein.get(
        normalize(anime.title.romaji),
        normalize(parsed.anime_title)
      );
      return distance < best.distance ? { parsed, distance } : best;
    }, { distance: Infinity }).parsed;

    if (bestMatch && bestMatch.episode_number) {
      const episodeData = await fetch(`${API_BASE_URL}/anime/${anime.idAnil}/${bestMatch.episode_number}`).then(res => res.json());
      const rssTorrent = rssData.find(anim => anim.title === bestMatch.file_name);
      return { ...anime, episode: episodeData, torrent: rssTorrent };
    }
    return null;
  }
  ```
  
  Esta parte por ejemplo, la funcion de reduce tiene su propio callback, entonces estan muchas cosas anidadas, eso aumenta la complejidad para leer porque tienes que en tu mente cuando estas leyendo este codigo, tener en cuenta:
  1. Devuelves un array de promesas
  2. Iteras sobre el array de `animes`
  3. En el array de `parsedAnimes` llamas un reducer para encontrar el `bestMatch`
  
  Entonces para entender la logica de este codigo, cargas con 3 cosas que uno como lector, tiene que tener en mente para entenderlo, es mejor irlas separando. En el primer ejemplo ya reducimos un poco la complejidad al separa lo de las promesas, ahora en el siguiente codigo, debemos de separar un poco 

 Tipo podemos crear una funcion que se encargue de una logica en concreto como decia antes, podemos crear la funcion `processAnimeWithBestMatch`, si te fijas el nombre de la funcion explica solita su proposito:

 ```js
 const processAnimeWithBestMatch = async (anime) => {
  const bestMatch = parsedAnimes.reduce((best, parsed) => {
    const distance = fastLevenshtein.get(
      normalize(anime.title.romaji),
      normalize(parsed.anime_title)
    );
    return distance < best.distance ? { parsed, distance } : best;
  }, { distance: Infinity }).parsed;

  if (bestMatch && bestMatch.episode_number) {
    const episodeResponse = await fetch(`${API_BASE_URL}/anime/${anime.idAnil}/${bestMatch.episode_number}`);
    const episodeData = await episodeResponse.json();
    const rssTorrent = rssData.find(anim => anim.title === bestMatch.file_name);
    return { ...anime, episode: episodeData, torrent: rssTorrent };
  }
  return null;
};
```

Ahora esta funcion esta un poco dificl de leer y entender, usar una funcion `reducer` es a veces complicada de leer porque  tienes que tener en mente que va a comparar los resultados y cada vez va a acumular el resultado, entonces ahi es importante separar un poquito la logica con `variables` y comentarios, por ejemplo:

1. Se actualizo `best` a `bestAnimeMatch` y `parsed` a `currentAnime`, para saber que es lo que estamos comparando el mejor resultado mejor anime match vs al actual que estamos por revisar.
2. Para no tener que encadenar con `.parsed` podemos usar la desestructuracion del objeto y luego renombrar el resultado `{ currentAnime: bestAnimeMatch }`, en lo personal en algunos casos no tan faciles de leer, no me gusta usar la encadenacion porque se puede volver dificil de leer y se nos puede pasar a la hora de leer codigo, aqui como esta ya muy complejo, es mejor hacerlo un poco mas claro
3. Tambien creo que tenias un error logico, leias al final de la reduccion `.parsed` cuando eso era el valor actual en la acumulacion, cuando deberias de leer al final `.best`, siendo el valor acumulado y por ende el mejor match.

Aqui esta toda la funcion corregida, pruebala a ver si funciona, si no debuggea que onda, pero ya es mucho mas legible y entender que hace la funcion:

```js
const processAnimes = async (rssData) => {
  const animeTitles = rssData.map(anime => anime.title);
  const parsedAnimes = await anitomyscript(animeTitles);

  const response = await fetch(`${API_BASE_URL}/anime/search`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ animes: parsedAnimes.map(a => a.anime_title) })
  });
  const animes = await response.json()

  const animeDataPromises = animes.map(animes.map(processAnimeWithBestMatch));

  const resolvedData = await Promise.all(animeDataPromises);

  return resolvedData.filter(Boolean);
}

const processAnimeWithBestMatch = async (anime) => {
  const bestAnimeMatch = parsedAnimes.reduce((bestMatch, currentAnime) => {
    const currentDistance = fastLevenshtein.get(
      normalize(anime.title.romaji),
      normalize(currentAnime.anime_title)
    );
    const isCurrentAnimeBetterMatch = currentDistance < bestMatch.distance;

    return isCurrentAnimeBetterMatch // If current anime is a better match
      ? { anime: currentAnime, distance: currentDistance } // Then update best match
      : bestMatch; // Otherwise keep current best match

  // Start with Infinity to ensure the first comparison always succeeds
  }, { anime: null, distance: Infinity });

  if (bestAnimeMatch.anime && bestAnimeMatch.anime.episode_number) {
    const episodeResponse = await fetch(`${API_BASE_URL}/anime/${anime.idAnil}/${bestAnimeMatch.anime.episode_number}`);
    const episodeData = await episodeResponse.json();
    const rssTorrent = rssData.find(anim => anim.title === bestAnimeMatch.anime.file_name);
    return { ...anime, episode: episodeData, torrent: rssTorrent };
  }

  return null;
};
```